File: backend/internal/domain/model/wod.go
package model

import "time"

type Wod struct {
	ID          string    `json:"id" firestore:"id"`
	Title       string    `json:"title" firestore:"title"`
	Description string    `json:"description" firestore:"description"`
	Date        time.Time `json:"date" firestore:"date"`
	CreatedAt   time.Time `json:"created_at" firestore:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" firestore:"updated_at"`
}

File: frontend/src/domain/entities/Wod.ts
export interface Wod {
  id: string;
  title: string;
  description: string;
  type: WodType;
  difficulty: WodDifficulty;
  movements: Movement[];
  scoring: ScoringType;
  timeLimit?: number;
  rounds?: number;
  createdAt: Date;
  updatedAt: Date;
}

export type WodType = 'amrap' | 'fortime' | 'emom' | 'tabata' | 'custom';

export type WodDifficulty = 'rx' | 'scaled' | 'beginner';

export interface Movement {
  name: string;
  reps: number;
  weight?: number;
  distance?: number;
  unit?: 'kg' | 'lb' | 'm' | 'cal';
  notes?: string;
}

export type ScoringType = {
  type: 'time' | 'reps' | 'weight' | 'rounds';
  value: number;
};
